base/ca/src/main/java/org/dogtagpki/server/ca/rest/base/ProfileBase.java

```diff
@@ -0,0 +1,1252 @@
+                ProfileConfig profileConfig = new ProfileConfig();
+                profileConfig.load(new ByteArrayInputStream(data));
+                tempProfile.init(engineConfig, registry, profileConfig);
+
+            } catch (Exception e) {
+                String message = "Unable to create profile: " + e.getMessage();
+                logger.error(message, e);
+                throw new BadRequestException(message, e);
+            }
+
+            // no error thrown, proceed with profile creation
+            profile = ps.createProfile(profileId, classId, className);
+            profile.getConfigStore().load(new ByteArrayInputStream(data));
+            ps.disableProfile(profileId);
+            ps.commitProfile(profileId);
+
+            auditProfileChange(
+                    ScopeDef.SC_PROFILE_RULES,
+                    OpDef.OP_ADD,
+                    profileId,
+                    ILogger.SUCCESS,
+                    auditParams);
+
+            return profileId;
+
+        } catch (EBaseException | IOException e) {
+
+            String message = "Unable to create profile: " + e.getMessage();
+            logger.error(message, e);
+
+            auditProfileChange(
+                    ScopeDef.SC_PROFILE_RULES,
+                    OpDef.OP_ADD,
+                    profileId,
+                    ILogger.FAILURE,
+                    auditParams);
+
+            throw new PKIException(message, e);
+        }
+    }
+
+    public void modifyProfileState(Principal principal, String profileId, String action) throws EBaseException {
+        if (profileId == null) {
+            logger.error("modifyProfileState: invalid request. profileId is null");
+            throw new BadRequestException("Unable to modify profile state: Invalid Profile Id");
+        }
+
+        if (action == null) {
+            logger.error("modifyProfileState: invalid request. action is null");
+            throw new BadRequestException("Unable to modify profile state: Missing action");
+        }
+
+        if (ps == null) {
+            logger.error("modifyProfileState: ps is null");
+            throw new PKIException("Error modifying profile state.  Profile Service not available");
+        }
+
+        try {
+            Profile profile = ps.getProfile(profileId);
+            if (profile == null) {
+                logger.error("Trying to modify profile: {}.  Profile not found.", profileId);
+                throw new ProfileNotFoundException(profileId);
+            }
+        } catch (EProfileException e1) {
+            e1.printStackTrace();
+            throw new PKIException("Error modifying profile state: unable to get profile");
+        }
+
+        switch (action) {
+        case "enable":
+            if (ps.isProfileEnable(profileId)) {
+                throw new ConflictingOperationException("Profile already enabled");
+            }
+            try {
+                ps.enableProfile(profileId, principal.getName());
```

Comment:
@edewata I know Marco is out on pto, so maybe you could help explain to me where the ACL for checking for the CA agent membership of the principal is held?

Reply:
Please see my comments here:
https://github.com/dogtagpki/pki/pull/4811#issuecomment-2253706306
https://github.com/dogtagpki/pki/pull/4811#discussion_r1697440313

So the original code checks whether the user is a member of `Certificate Manager Agents` or `Certificate Manager Administrators` directly, and the new code checks whether the user has `read` access to `certServer.profile.configuration` using ACL, but IIUC they are not identical replacements, that's my concern with this change.

Reply:
thanks @edewata .  So, with the clarification that this task is to replace the rest interface with the same policy, I agree with Endi that the code in this patch here isn't identical.  Although I have to add that as I stated earlier, only CA agents should have the privilege to enable or disable a profile.  I'll leave it to you two to decide whether to change it to the original behavior for this patch now, then create a separate ticket to change it to the desired behavior I stated, or simply change it to match the desired behavior.